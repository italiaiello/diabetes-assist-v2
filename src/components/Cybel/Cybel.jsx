import React from 'react'
import Technician from '../../images/technician.svg'
import Microphone from '../../images/microphone.svg'
import HomeButton from '../../images/Home.svg'

const Cybel = props => {
    return (
        <div className="pageDisplay">
            <div className="backAndHeading">
                <div className="sectionHeading">
                    <div className="back">
                        <div className="leftArrow" onClick={() => props.onRouteChange('home')}></div>
                        <p>Back</p>
                    </div>
                </div>
            </div>
            <figure id="cybelRobot">
                <img src={Technician} alt="Cybel the robot" />
            </figure>
            <p>Talk to me about anything and I can give you advice
                <br />
                <br />
                Go ahead, I&#39;m listening...
            </p>
            <figure id="microphone">
                <img src={Microphone} alt="Voice command microphone" />
            </figure>
            <figure className="homeButton">
                <img id="homeCybel" src={HomeButton} alt="Home button" onClick={() => props.onRouteChange('home')} />
            </figure>
        </div>
    )
}

export default Cybel