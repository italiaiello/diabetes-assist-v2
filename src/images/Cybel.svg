<svg width="82" height="54" viewBox="0 0 82 54" fill="none" xmlns="http://www.w3.org/2000/svg">
<g filter="url(#filter0_d)">
<g clip-path="url(#clip0)">
<path d="M42.418 7.58422C33.6402 -1.19439 19.359 -1.19439 10.5829 7.5821C1.80514 16.3586 1.80599 30.6403 10.5837 39.4184C19.3589 48.1945 33.6402 48.1945 42.4171 39.4167C51.194 30.6403 51.1936 16.3594 42.418 7.58422ZM39.9617 36.9622C32.5392 44.3848 20.4613 44.3856 13.0383 36.9631C5.61401 29.5392 5.61486 17.46 13.0383 10.0375C20.4608 2.61533 32.5383 2.61449 39.9626 10.0383C47.3851 17.4609 47.3843 29.5401 39.9617 36.9622ZM18.1035 17.4079C18.1035 15.9551 19.2817 14.7768 20.7345 14.7768C22.1869 14.7768 23.3651 15.9547 23.3651 17.4079C23.3651 18.8615 22.1869 20.0393 20.7345 20.0393C19.2817 20.0393 18.1035 18.8615 18.1035 17.4079ZM30.0333 17.4079C30.0333 15.9551 31.2124 14.7768 32.6652 14.7768C34.1175 14.7768 35.2958 15.9547 35.2958 17.4079C35.2958 18.8615 34.118 20.0393 32.6652 20.0393C31.2124 20.0393 30.0333 18.8615 30.0333 17.4079ZM36.2534 28.1836C34.6229 31.9543 30.8042 34.3906 26.5257 34.3906C22.155 34.3906 18.3139 31.942 16.7398 28.1522C16.4699 27.5035 16.7776 26.7584 17.4271 26.4886C17.5867 26.4228 17.7522 26.3914 17.9151 26.3914C18.4141 26.3914 18.8876 26.6863 19.0908 27.1763C20.2691 30.0132 23.1873 31.8452 26.5257 31.8452C29.7885 31.8452 32.6898 30.0119 33.9168 27.1734C34.196 26.528 34.9453 26.2306 35.5902 26.5102C36.2352 26.7898 36.5326 27.5387 36.2534 28.1836Z" fill="white"/>
</g>
</g>
<g filter="url(#filter1_d)">
<path d="M78.0001 18.2728C78.0001 14.9223 75.9356 11.9321 72.8436 10.4775C72.7475 16.9017 67.1534 22.11 60.2534 22.1994C61.8157 25.0782 65.0274 27.0003 68.6261 27.0003C70.3133 27.0003 71.9541 26.5819 73.3958 25.787L77.9591 26.9622L76.6969 22.7136C77.5507 21.3713 78.0001 19.8437 78.0001 18.2728Z" fill="white"/>
<path d="M71.1465 10.3096C71.1465 4.62476 66.1791 0 60.0732 0C53.9673 0 49 4.62476 49 10.3096C49 12.1623 49.5297 13.9662 50.5355 15.5498L49.0407 20.581L54.4448 19.1895C56.1458 20.126 58.0833 20.6191 60.0732 20.6191C66.1791 20.6191 71.1465 15.9944 71.1465 10.3096ZM58.374 7.91016H56.6748C56.6748 6.16539 58.1992 4.74609 60.0732 4.74609C61.9473 4.74609 63.4717 6.16539 63.4717 7.91016C63.4717 8.79572 63.069 9.64689 62.3665 10.2451L60.9229 11.4753V12.709H59.2236V10.7786L61.2193 9.07793C61.576 8.77409 61.7725 8.35943 61.7725 7.91016C61.7725 7.03777 61.0102 6.32812 60.0732 6.32812C59.1362 6.32812 58.374 7.03777 58.374 7.91016ZM59.2236 14.291H60.9229V15.873H59.2236V14.291Z" fill="white"/>
</g>
<defs>
<filter id="filter0_d" x="0" y="1" width="53" height="53" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
<feFlood flood-opacity="0" result="BackgroundImageFix"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/>
<feOffset dy="4"/>
<feGaussianBlur stdDeviation="2"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
<feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
</filter>
<filter id="filter1_d" x="45" y="0" width="37" height="35" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
<feFlood flood-opacity="0" result="BackgroundImageFix"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/>
<feOffset dy="4"/>
<feGaussianBlur stdDeviation="2"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
<feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
</filter>
<clipPath id="clip0">
<rect x="4" y="1" width="45" height="45" fill="white"/>
</clipPath>
</defs>
</svg>
