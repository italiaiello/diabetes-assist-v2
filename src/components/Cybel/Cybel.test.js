import { shallow } from 'enzyme';
import React from 'react';
import Cybel from './Cybel';

let route = ""

const mockOnRouteChange = (newRoute) => {
  route = newRoute;
}

describe('Cybel.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Cybel onRouteChange={mockOnRouteChange} />)
  })

  it('expects to render Cybel component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('checks that onRouteChange works onClick', () => {
    wrapper.find('[id="homeCybel"]').simulate('click')
    expect(route).toEqual('home')

    route = "chicken"

    wrapper.find('[className="leftArrow"]').simulate('click')
    expect(route).toEqual('home')
  })
}) 