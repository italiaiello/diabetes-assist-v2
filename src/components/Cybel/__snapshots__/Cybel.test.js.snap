// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cybel.js expects to render Cybel component 1`] = `ShallowWrapper {}`;
